src/main/java/io/sc3/plethora/mixin/EntityMixin.java

```diff
@@ -0,0 +1,26 @@
+package io.sc3.plethora.mixin;
+
+import io.sc3.plethora.util.VelocityDeterminable;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.Vec3d;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+@Mixin(Entity.class)
+public class EntityMixin implements VelocityDeterminable {
+    @Unique
+    public Vec3d prevPos = Vec3d.ZERO;
+    public Vec3d motion = Vec3d.ZERO;
```

Comment:
Don't forget @Unique here too
```suggestion
    @Unique
    private Vec3d motion = Vec3d.ZERO;
```